package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -972,7 +972,12 @@ dnsmasq_start()
 	config_list_foreach "$cfg" "addnhosts" append_addnhosts
 	config_list_foreach "$cfg" "bogusnxdomain" append_bogusnxdomain
 	append_parm "$cfg" "leasefile" "--dhcp-leasefile" "/tmp/dhcp.leases"
-	append_parm "$cfg" "serversfile" "--servers-file"
+	local serversfile
+	config_get serversfile "$cfg" "serversfile"
+	if [ -n "$serversfile" ]; then
+		xappend "--servers-file=$serversfile"
+		append EXTRA_MOUNT "$(dirname "$serversfile")"
```

Comment:
I can still see that people will set this to `/tmp/serversfile` and that will break jailing by mounting `/tmp` read-only.
Maybe
```suggestion
	local serversfile_dir="$(dirname "$serversfile")"
	if [ "$serversfile_dir" = "/tmp" ]; then
		log_once "Please move serversfile to a directory other than /tmp if it is going to be replaced"
		append EXTRA_MOUNT "$serversfile"
	else
		append EXTRA_MOUNT "$serversfile_dir"
	fi
```

Reply:
Would it not break because it is the same (default) dir as `dhcp.leases` which needs to be rw mounted, rather than because it's in `/tmp`?

Reply:
@clayface Exactly. Without my suggestion which would handle that case of serversfile loacted in `/tmp`, this would be the problem. Not saying that it's the best way to deal with it, but it would at least prevent the most obvious misconfiguration.

Reply:
@dangowrt do you want me to implement your fix and force-push an update or can you pick the change including your fix for `/tmp` when committing to git.openwrt.org?

Reply:
@dangowrt I have added a workaround for this on #4864 that should take care of any RO/RW conflicts. Can you review?